lib/iris/fileformats/netcdf.py

```diff
@@ -780,6 +784,13 @@ def write(self, cube, local_keys=None, unlimited_dimensions=None,
                     'cf_patch')
                 warnings.warn(msg)
 
+        # Perform any user registered callback function.
+        # As the NetCDF saver commits to disk as it processes the cube and
+        # its metadata, it is not possible to skip or ignore the derived
+        # NetCDF representation of a cube as with other file format savers.
```

Comment:
Also, the `field` terminology doesn't really fit for netCDF. Which begs the question, "What does the callback routine get called with?" I'd assume a netCDF `Variable`, but that would need to be documented. Unfortunately, it wouldn't be very nice to document it on `iris.save()`.